app/models/conversion_host/configurations.rb

```diff
@@ -24,7 +25,21 @@ def queue_configuration(op, instance_id, resource, params, auth_user = nil)
         :zone        => resource.ext_management_system.my_zone,
         :args        => params
       }
-      MiqTask.generic_action_with_callback(task_opts, queue_opts)
+
+      task_id = MiqTask.generic_action_with_callback(task_opts, queue_opts)
```

Comment:
@djberg96 how about adding `:context_data` to the [options hash](https://github.com/ManageIQ/manageiq/blob/a61936979cca681335170c252cfdee2a8f26b6e4/app/models/miq_task.rb#L264) and pass it to the `MiqTask.create!` [here](https://github.com/ManageIQ/manageiq/blob/a61936979cca681335170c252cfdee2a8f26b6e4/app/models/miq_task.rb#L286)

Reply:
I'd rather fix it first even if it is in a separate PR, a quick search of the callers should be enough to see if anyone is doing anything strange like passing in invalid keys.

I'd expect to be able to control any of the task attributes in the first argument, so something like:
```
Task.create!(
  {
      :state   => STATE_QUEUED,
      :status  => STATUS_OK,
      :message => msg
  }.merge(options)
)
```

Reply:
I actually had this already started, so I've created a WIP here: https://github.com/ManageIQ/manageiq/pull/18578

Reply:
Awesome!